pkg/cloud/common/sources.go

```diff
@@ -8,11 +8,17 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+// ObjectSource is a holder for the path of the object to read from
+// and typed Object, where the resource content would be unserialized
 type ObjectSource struct {
 	Object client.Object
 	Path   string
 }
 
+// ReadResources is responsible for reading resources from ebmed path in the binary.
+// This method is for internal use only, and a <provider> should use this method
+// only in init() and later provide a copy of aquired objects. See /pkg/cloud for
```

Comment:
i'm curious why this needs to be run in an init() method?

Reply:
`init()` will run only once, and a failure there will prevent the binary from starting, which will be immediately displayed in CI tests and in attempts to run CCCMO locally. As we don't require any external dependency, and all our manifests are baked in the binary with embed, an ideal scenario would be that this read would not create any errors. Unfortunately we can't rely on that, so we need to make sure that the implementation would return the error if there is an issue with this static manifest. There could be no changes to this manifest after it, so the moment it being added to the binary is the only one we need to watch. Ideal scenario for init.

Reply:
perfect, thank you for the explanation =)